qiskit/pulse/schedule.py

```diff
@@ -517,6 +517,134 @@ def _add_timeslots(self, time: int, schedule: ScheduleComponent) -> None:
+                    "channel {ch} in Schedule(name='{name}').".format(
+                        ch=channel, t0=interval[0], tf=interval[1], name=schedule.name))
+
+            if not channel_timeslots:
+                self._timeslots.pop(channel)
+
+    def _replace_timeslots(self,
+                           time: int,
+                           old: ScheduleComponent,
+                           new: ScheduleComponent):
+        """Replace the timeslots of ``old`` if present with the timeslots of ``new``.
+
+        Args:
+            time: The time to remove the timeslots for the ``schedule`` component.
+            old: Instruction to replace.
+            new: Instruction to replace with.
+        """
+        self._remove_timeslots(time, old)
+        self._add_timeslots(time, new)
+
+    def replace(self,
+                old: ScheduleComponent,
+                new: ScheduleComponent,
+                inplace: bool = False,
+                ) -> 'Schedule':
+        """Return a schedule with the ``old`` instruction replaced with a ``new``
+        instruction.
+
+        The replacment matching is based on an instruction equality check.
+
+        .. jupyter-kernel:: python3
+          :id: replace
+
+        .. jupyter-execute::
+
+          from qiskit import pulse
+
+          d0 = pulse.DriveChannel(0)
+
+          sched = pulse.Schedule()
+
+          old = pulse.Play(pulse.Constant(100, 1.0), d0)
+          new = pulse.Play(pulse.Constant(100, 0.1), d0)
+
+          sched += old
+
+          sched = sched.replace(old, new)
+
+          assert sched == pulse.Schedule(new)
+
+        Only matches at the top-level of the schedule tree. If you wish to
+        perform this replacement over all instructions in the schedule tree.
+        Flatten the schedule prior to running::
+
+        .. jupyter-execute::
+
+          sched = pulse.Schedule()
+
+          sched += pulse.Schedule(old)
+
+          sched = sched.flatten()
+
+          sched = sched.replace(old, new)
+
+          assert sched == pulse.Schedule(new)
+
+        Args:
+          old: Instruction to replace.
+          new: Instruction to replace with.
+          inplace: Replace instruction by mutably modifying this ``Schedule``.
+
+        Returns:
+          The modified schedule with ``old`` replaced by ``new``.
+
+        Raises:
+            PulseError: If the ``Schedule`` after replacements will has a timing overlap.
+        """
+        new_children = []
+        for time, child in self._children:
+            if child == old:
+                new_children.append((time, new))
+                if inplace:
+                    self._replace_timeslots(time, old, new)
+            else:
+                new_children.append((time, child))
+
+        try:
+            if inplace:
+                self.__children = new_children
+                return self
+            else:
+                return Schedule(*new_children)
```

Comment:
Is it this line that would cause the PulseError? Could this be:
```
if ...:
   ...
else:
   try:
      return S
   except:
```

Reply:
Fixed in b727d727b.